panel/widgets/button.py

```diff
@@ -243,17 +243,26 @@ def _process_event(self, event: param.parameterized.Event) -> None:
         self.clicks += 1
 
     def on_click(
-        self, callback: Callable[[param.parameterized.Event], None]
+        self, callback: Callable[[param.parameterized.Event], None | Awaitable[None]]
```

Comment:
This change is important for you to notice @philippjfr . Async functions need special `Awaitable` types.

One day we should run `mypy` on the codebase. Would identify a bunch of potential issues, that if fixed would avoid exceptions or help provide better information/ auto-completion in editors.

Reply:
Thanks, and yes, I do want to add mypy to the automated code checks eventually.